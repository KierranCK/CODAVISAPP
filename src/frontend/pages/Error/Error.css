.launchpad {
  height: 100%;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  position: relative;
}

.launchpad-title-pane {
  z-index: 10;
  height: 100%;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
  position: relative;
  left: -7.8em;
  padding-left: 5em;
  width: 100%;
  -webkit-transform: translateX(100px);
          transform: translateX(100px);
  height: 100vh;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
  gap: 0.8em;
  overflow: hidden;
}

.launchpad-title-pane::after {
  content: "";
  background: -webkit-gradient(linear, left top, right top, from(transparent), color-stop(75%, black), to(transparent));
  background: linear-gradient(90deg, transparent, black 75%, transparent);
  pointer-events: none;
}

.visualise-image img {
  height: 100vh;
  z-index: -10;
  right: 0;
}

button {
  padding: 0.5em 2.5em !important;
}
/*# sourceMappingURL=Error.css.map */