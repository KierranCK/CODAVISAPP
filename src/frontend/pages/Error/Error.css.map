{
    "version": 3,
    "mappings": "AAAA,AAAA,UAAU,CAAC;EACP,MAAM,EAAE,IAAI;EACZ,OAAO,EAAE,IAAI;EACb,QAAQ,EAAE,QAAQ;CACnB;;AAED,AAAA,qBAAqB,CAAC;EACpB,OAAO,EAAE,EAAE;EACX,MAAM,EAAE,IAAI;EACZ,OAAO,EAAE,IAAI;EACb,cAAc,EAAE,MAAM;EACtB,QAAQ,EAAE,QAAQ;EAClB,IAAI,EAAE,MAAM;EACZ,YAAY,EAAE,GAAG;EACjB,KAAK,EAAE,IAAI;EACX,SAAS,EAAE,iBAAiB;EAC5B,MAAM,EAAE,KAAK;EACb,eAAe,EAAE,MAAM;EACvB,GAAG,EAAE,KAAK;EACV,QAAQ,EAAE,MAAM;CAYjB;;AAzBD,AAeE,qBAfmB,AAelB,OAAO,CAAC;EACP,OAAO,EAAE,EAAE;EACb,UAAU,EAAE,2DAKX;EACC,cAAc,EAAE,IAAI;CACrB;;AAGH,AAAA,gBAAgB,CAAC,GAAG,CAAC;EACnB,MAAM,EAAE,KAAK;EACb,OAAO,EAAE,GAAG;EACZ,KAAK,EAAE,CAAC;CACT;;AAED,AAAA,MAAM,CAAC;EACL,OAAO,EAAE,sBAAsB;CAChC",
    "sources": [
        "Error.scss"
    ],
    "names": [],
    "file": "Error.css"
}