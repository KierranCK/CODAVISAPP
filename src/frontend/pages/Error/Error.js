import React from 'react'
import './Error.css'
import { Row, Col, Button } from 'react-bootstrap'
import { Link } from 'react-router-dom'

const Error = () => {
  return (
    <>
        <div className='launchpad-title-pane'>
        <h1>404</h1>
        <p>Oops! Page not Found</p>

        <Row>
          <Col>
          <Link to="/dashboard">
          <Button variant="warning" size="sm">Dashboard</Button>
          </Link>

          </Col>
        </Row>

        </div>
    </>
  )
}

export default Error