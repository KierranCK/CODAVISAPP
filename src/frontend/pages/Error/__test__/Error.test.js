import {
  render,
  screen,
  cleanup,
  fireEvent,
  waitFor,
  waitForElementToBeRemoved,
} from "@testing-library/react";
import React from "react";
import { BrowserRouter } from "react-router-dom";
import Error from "../Error";

//function to render component to be tested
const renderComponent = () =>
  render(<Error />, {
    wrapper: BrowserRouter,
  });

//object containing all elements to be tested
const elements = {
  heading: () => screen.getByRole("heading"),
  button: () => screen.getByRole("button"),
};

//function to render component before each test
beforeEach(() => {
  renderComponent();
});

//function cleanup after each test
afterEach(() => {
  cleanup();
});

//test suite
describe("Error Tests", () => {
  test("Ensure Error has not inadvertently changed", () => {
    expect(renderComponent().asFragment()).toMatchSnapshot();
  });
  test('Ensure heading and "Dashboard" button rendered', () => {
    expect(elements.heading()).toHaveTextContent("404");
    expect(elements.button()).toBeInTheDocument();
    expect(elements.button()).toHaveTextContent("Dashboard");
    expect(elements.button()).not.toBeDisabled();
  });
});
