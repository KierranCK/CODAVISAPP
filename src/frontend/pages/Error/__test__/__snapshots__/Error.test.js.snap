// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Tests Ensure Launchpad has not inadvertently changed 1`] = `
<DocumentFragment>
  <div
    class="launchpad-title-pane"
  >
    <h1>
      404
    </h1>
    <p>
      Oops! Page not Found
    </p>
    <div
      class="row"
    >
      <div
        class="col"
      >
        <a
          href="/dashboard"
        >
          <button
            class="btn btn-warning btn-sm"
            type="button"
          >
            Dashboard
          </button>
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Error Tests Ensure Error has not inadvertently changed 1`] = `
<DocumentFragment>
  <div
    class="launchpad-title-pane"
  >
    <h1>
      404
    </h1>
    <p>
      Oops! Page not Found
    </p>
    <div
      class="row"
    >
      <div
        class="col"
      >
        <a
          href="/dashboard"
        >
          <button
            class="btn btn-warning btn-sm"
            type="button"
          >
            Dashboard
          </button>
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Launchpad Tests Ensure Launchpad has not inadvertently changed 1`] = `
<DocumentFragment>
  <div
    class="launchpad-title-pane"
  >
    <h1>
      404
    </h1>
    <p>
      Oops! Page not Found
    </p>
    <div
      class="row"
    >
      <div
        class="col"
      >
        <a
          href="/dashboard"
        >
          <button
            class="btn btn-warning btn-sm"
            type="button"
          >
            Dashboard
          </button>
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
